test/utils.jl

```diff
@@ -38,19 +38,38 @@ end
     if Sys.iswindows()
         @test Atom.fullREPLpath(raw"@ Atom C:\Users\ads\.julia\dev\Atom\src\repl.jl:25") == (raw"C:\Users\ads\.julia\dev\Atom\src\repl.jl", 25)
-        @test Atom.fullREPLpath(raw"@ Atom C:\Users\ads one\.julia\dev\Atom\src\repl.jl:25") == (raw"C:\Users\ads one\.julia\dev\Atom\src\repl.jl", 25)
-        @test Atom.fullREPLpath(raw"C:\Users\ads\.julia\dev\Atom\src\repl.jl:25") == (raw"C:\Users\ads\.julia\dev\Atom\src\repl.jl", 25)
-        @test Atom.fullREPLpath(raw"C:\Users\ads one\.julia\dev\Atom\src\repl.jl:25") == (raw"C:\Users\ads one\.julia\dev\Atom\src\repl.jl", 25)
-        @test Atom.fullREPLpath(raw".\foo\bar.jl:1") == (Atom.fullpath(raw".\foo\bar.jl"), 1)
-        @test Atom.fullREPLpath(raw".\fo o\bar.jl:1") == (Atom.fullpath(raw".\fo o\bar.jl"), 1)
-        @test Atom.fullREPLpath(raw".\foo\ba r.jl:1") == (Atom.fullpath(raw".\foo\ba r.jl"), 1)
-        @test Atom.fullREPLpath(raw"foo\bar.jl:1") == (Atom.fullpath(raw".\foo\bar.jl"), 1)
-        @test Atom.fullREPLpath(raw"foo\ba r.jl:1") == (Atom.fullpath(raw".\foo\ba r.jl"), 1)
+        @test Atom.fullREPLpath(raw"@ Atom C:\Users\ads\.julia\dev\At om\src\repl.jl:25") == (raw"C:\Users\ads\.julia\dev\At om\src\repl.jl", 25)
+        @test Atom.fullREPLpath(raw"@ Atom Users\ads\.julia\dev\At om\src\repl.jl:25") == (raw"Users\ads\.julia\dev\At om\src\repl.jl", 25)
+
+        @test Atom.fullREPLpath(raw"C:\Us ers\ads\.julia\dev\At om\src\re pl.jl:25") == (raw"C:\Us ers\ads\.julia\dev\At om\src\re pl.jl", 25)
+        @test Atom.fullREPLpath(raw"C:\Us ers\ads\.julia\dev\At om\src\repl:25") == (raw"C:\Us ers\ads\.julia\dev\At om\src\repl", 25)
+        @test Atom.fullREPLpath(raw"C:\Us ers\ads\.julia\dev\At om\src\repl") == (raw"C:\Us ers\ads\.julia\dev\At om\src\repl", 0)
+        @test Atom.fullREPLpath(raw"C:\Us ers\ads\.julia\dev\At om\src\repl.jl") == (raw"C:\Us ers\ads\.julia\dev\At om\src\repl.jl", 0)
+        @test Atom.fullREPLpath(raw"C:\Us ers\ads one\julia\dev\At om\src\repl.jl") == (raw"C:\Us ers\ads one\julia\dev\At om\src\repl.jl", 0)
+        @test Atom.fullREPLpath(raw"C:/Us ers/ads one/julia/dev/At om\src/re pl.jl") == (raw"C:/Us ers/ads one/julia/dev/At om\src/re pl.jl", 0)
+
+        @test Atom.fullREPLpath(raw"Users\ads\.julia\dev\Atom\src\repl.jl:25") == (Atom.fullpath(raw"Users\ads\.julia\dev\Atom\src\repl.jl"), 25)
+        @test Atom.fullREPLpath(raw"Users\ads\.julia\dev\At om\src\repl.jl:25") == (Atom.fullpath(raw"Users\ads\.julia\dev\At om\src\repl.jl"), 25)
+        @test Atom.fullREPLpath(raw"Users\ads\.julia\dev\At om\src\repl.jl:25") == (Atom.fullpath(raw".\Users\ads\.julia\dev\At om\src\repl.jl"), 25)
+        @test Atom.fullREPLpath(raw".\Us ers\ads\.julia\dev\At om\src\repl.jl:25") == (Atom.fullpath(raw".\Us ers\ads\.julia\dev\At om\src\repl.jl"), 25)
+        @test Atom.fullREPLpath(raw"..\Us ers\ads\.julia\dev\At om\src\repl.jl:25") == (Atom.fullpath(raw"..\Us ers\ads\.julia\dev\At om\src\repl.jl"), 25)
```

Comment:
What are you trying to test here ?
both `fullREPLpath` and `fullpath` try to "expand" a short path:
- https://github.com/JunoLab/Atom.jl/blob/40f5c6007c0511d09f39020c232ce4087c1bd5a4/src/utils.jl#L74-L78
- https://github.com/JunoLab/Atom.jl/blob/master/src/path_matching.jl#L18
but that doesn't happen for those full-like paths.

If you want to test our regex correctly matches relative paths, something like
```julia
# test fullpath
@test Atom.fullREPLpath(raw"Users\ads\.julia\dev\Atom\src\repl.jl:25") == (raw"Users\ads\.julia\dev\Atom\src\repl.jl", 25)
@test Atom.fullREPLpath(raw"Users\ads\.julia\dev\Atom\sr c\repl.jl:25") == (raw"Users\ads\.julia\dev\Atom\sr c\repl.jl", 25)
@test Atom.fullREPLpath(raw"Users\ads\.julia\dev\Atom\src\re pl.jl:25") == (raw".\Users\ads\.julia\dev\Atom\src\re pl.jl", 25)

# test relative path
@test Atom.fullREPLpath("../foo/bar.jl") == (Atom.fullpath("../foo/bar.jl"), 0)
@test Atom.fullREPLpath("../foo/bar.jl:10") == (Atom.fullpath("../foo/bar.jl"), 10)
@test Atom.fullREPLpath("../fo o/bar.jl") == (Atom.fullpath("../fo o/bar.jl"), 0)
@test Atom.fullREPLpath("../foo/b ar.jl:10") == (Atom.fullpath("../foo/b ar.jl"), 10)
```
might be more realistic and make more sense. 
Same goes for UNIX system test.

Reply:
I'll merge this for now, we can always improve tests in a follow up PR.

Reply:
okay, yeah, speeding up development by making it easy to contribute is much more important than enriching tests.

Thanks @FAlobaid for your help !

Reply:
Thanks @aviatesk and @pfitzseb. Very happy to help 😃.

Just a quick note about the test cases. @aviatesk your suggested full path tests above will actually fail as `fullREPLpath` will expand any path that doesn't start with `\` or `/` or `[driveletter]:`. I know my relative path tests are deceptively long, this is to make the test cases robust having different combinations of characters `\ / .` and spaces.

I actually included 3 test cases for the `@` path pattern, 6 for full paths and 5 for relative paths for Win / macOS / Linux. And as @pfitzseb said, we can always improve it later as you guys see fit 😃. 

Thanks and keep up the good work :+1: